### PR TITLE
resource_manager_client: fix client nil possible to fix panic 

### LINKED ISSUE #6131: tso: pd nit pointer panic in tso dispatcher
### What did you expect to see?
normal 
### What did you see instead?
[panic](https://github.com/tikv/pd/actions/runs/4373197579/jobs/7651070154)
```
2023-03-09T10:05:14.8268865Z [2023/03/09 10:05:13.501 +00:00] [INFO] [client.go:290] ["[pd] create pd client with endpoints"] [pd-address="[https://127.0.0.1:36315]"]
2023-03-09T10:05:14.8269328Z [2023/03/09 10:05:13.554 +00:00] [INFO] [base_client.go:480] ["[pd] switch leader"] [new-leader=https://127.0.0.1:36315] [old-leader=]
2023-03-09T10:05:14.8269675Z [2023/03/09 10:05:13.554 +00:00] [INFO] [base_client.go:156] ["[pd] init cluster id"] [cluster-id=7208485466879229826]
2023-03-09T10:05:14.8270049Z [2023/03/09 10:05:13.555 +00:00] [INFO] [tso_request_dispatcher.go:287] ["[pd/tso] tso dispatcher created"] [dc-location=global]
2023-03-09T10:05:14.8270446Z [2023/03/09 10:05:13.555 +00:00] [INFO] [client.go:290] ["[pd] create pd client with endpoints"] [pd-address="[https://127.0.0.1:36315]"]
2023-03-09T10:05:14.8270786Z [2023/03/09 10:05:13.562 +00:00] [ERROR] [tso_request_dispatcher.go:507] ["[pd/tso] update connection contexts failed"] [dc=global] [error="rpc error: code = Canceled desc = context canceled"]
2023-03-09T10:05:14.8271151Z [2023/03/09 10:05:13.562 +00:00] [INFO] [tso_request_dispatcher.go:305] ["[pd/tso] exit tso dispatcher"] [dc-location=global]
2023-03-09T10:05:14.8271355Z panic: runtime error: invalid memory address or nil pointer dereference
2023-03-09T10:05:14.8271548Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1eedf59]
2023-03-09T10:05:14.8271566Z 
2023-03-09T10:05:14.8271675Z goroutine 348500 [running]:
2023-03-09T10:05:14.8271933Z github.com/tikv/pd/client.(*client).tryResourceManagerConnect(0xc007054720?, {0x554f930, 0xc00a167bd0}, 0xc0057c1e58)
2023-03-09T10:05:14.8272146Z 	/home/runner/work/pd/pd/client/resource_manager_client.go:354 +0x119
2023-03-09T10:05:14.8272414Z github.com/tikv/pd/client.(*client).handleResourceTokenDispatcher(0xc015a9ae10, {0x554f930, 0xc00a167bd0}, 0xc015bdd300)
2023-03-09T10:05:14.8272628Z 	/home/runner/work/pd/pd/client/resource_manager_client.go:278 +0xdb
2023-03-09T10:05:14.8272820Z created by github.com/tikv/pd/client.(*client).createTokenDispatcher
2023-03-09T10:05:14.8273015Z 	/home/runner/work/pd/pd/client/resource_manager_client.go:265 +0x2cf
2023-03-09T10:05:14.8273162Z FAIL	github.com/tikv/pd/tests/client	230.444s
2023-03-09T10:05:14.8273243Z FAIL
```
### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Close #6131 Ref #5851

### What is changed and how does it work?

```commit-message
need to check GetOrCreateGRPCConn error to avoid panic
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by HuSharp** [comment]
@CabinfeverB @bufferflies  PTAL thx!

**Review by nolouch — APPROVED** [review]
lgtm

File: client/resource_manager_client.go [hunk 1]
```diff
@@ -67,7 +68,11 @@ func (c *client) gRPCErrorHandler(err error) {
 // ListResourceGroups loads and returns all metadata of resource groups.
 func (c *client) ListResourceGroups(ctx context.Context) ([]*rmpb.ResourceGroup, error) {
 	req := &rmpb.ListResourceGroupsRequest{}
```
Comment (JmPotato) [inline client/resource_manager_client.go:70]: We can create the request struct after `c.resourceManagerClient()`.

File: client/resource_manager_client.go [hunk 2]
```diff
@@ -103,34 +112,41 @@ func (c *client) ModifyResourceGroup(ctx context.Context, metaGroup *rmpb.Resour
 	return c.putResourceGroup(ctx, metaGroup, modify)
 }
 
-func (c *client) putResourceGroup(ctx context.Context, metaGroup *rmpb.ResourceGroup, typ actionType) (str string, err error) {
+func (c *client) putResourceGroup(ctx context.Context, metaGroup *rmpb.ResourceGroup, typ actionType) (string, error) {
 	req := &rmpb.PutResourceGroupRequest{
 		Group: metaGroup,
 	}
 	var resp *rmpb.PutResourceGroupResponse
```
Comment (JmPotato) [inline client/resource_manager_client.go:119]: Ditto.

File: client/resource_manager_client.go (hunk 1 above)
Comment (HuSharp) [reply client/resource_manager_client.go]: done

**Review by JmPotato — APPROVED** [review]

**Comment by HuSharp** [comment]
@JmPotato can u add a `ok-to-test` label, thx!

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**